src/main/java/com/radixdlt/crypto/HashUtils.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.crypto;
+
+import com.google.common.hash.HashCode;
+import com.google.common.primitives.UnsignedBytes;
+import com.radixdlt.SecurityCritical;
+import com.radixdlt.SecurityCritical.SecurityKind;
+
+import java.security.SecureRandom;
+import java.util.Comparator;
+
+@SecurityCritical(SecurityKind.HASHING)
+public final class HashUtils {

```

Comment:
Add javadoc, would also like to see a comment that static methods should get replaced by instance methods at some point.

Reply:
what's the reason for using instance methods rather than static here? if you think it's worth it I'd rather do it straight away than leave a comment

Reply:
static class doesn't allow one to change implementations, e.g. if we need to mock hashing and use a cheaper version of hashing for testing, or add the ability to cache.

I don't think it's worth it now because our codebase is currently spread with a lot of these static calls and would need to see what the actual correct interface is when combined with our use of our Hasher interface class.